python/hail/expr/tests.py

```diff
@@ -192,3 +192,7 @@ def assert_typed(expr, result, dtype):
         assert_typed(s.annotate(),
                      Struct(f1=1, f2=2, f3=3),
                      TStruct(['f1', 'f2', 'f3'], [TInt32(),TInt32(),TInt32()]))
+
+    def test_lambdas(self):
+        a = functions.capture([1,2,3])
+        self.assertRaises(TypeError, lambda: eval_expr(list(a.map(lambda x: x * 2))))
```

Comment:
Why not just `eval_expr(list(a))`?

Reply:
that would also trigger it. I had this from when I thought it was related to the lambda stuff

Reply:
Your call